pkg/apis/core/v1beta1/types_cloudprofile.go

```diff
@@ -0,0 +1,174 @@
+	Items []CloudProfile `json:"items"`
+}
+
+// CloudProfileSpec is the specification of a CloudProfile.
+// It must contain exactly one of its defined keys.
+type CloudProfileSpec struct {
+	// CABundle is a certificate bundle which will be installed onto every host machine of shoot cluster targetting this profile.
+	// +optional
+	CABundle *string `json:"caBundle,omitempty"`
+	// Kubernetes contains constraints regarding allowed values of the 'kubernetes' block in the Shoot specification.
+	Kubernetes KubernetesSettings `json:"kubernetes"`
+	// MachineImages contains constraints regarding allowed values for machine images in the Shoot specification.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	MachineImages []MachineImage `json:"machineImages" patchStrategy:"merge" patchMergeKey:"name"`
+	// MachineTypes contains constraints regarding allowed values for machine types in the 'workers' block in the Shoot specification.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	MachineTypes []MachineType `json:"machineTypes" patchStrategy:"merge" patchMergeKey:"name"`
+	// ProviderConfig contains provider-specific configuration for the profile.
+	// +optional
+	ProviderConfig *ProviderConfig `json:"providerConfig,omitempty"`
+	// Regions contains constraints regarding allowed values for regions and zones.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	Regions []Region `json:"regions" patchStrategy:"merge" patchMergeKey:"name"`
+	// SeedSelector contains an optional list of labels on `Seed` resources that marks those seeds whose shoots may use this provider profile.
+	// An empty list means that all seeds of the same provider type are supported.
+	// This is useful for environments that are of the same type (like openstack) but may have different "instances"/landscapes.
+	// +optional
+	SeedSelector *metav1.LabelSelector `json:"seedSelector,omitempty"`
+	// Type is the name of the provider.
+	Type string `json:"type"`
+	// VolumeTypes contains constraints regarding allowed values for volume types in the 'workers' block in the Shoot specification.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +optional
+	VolumeTypes []VolumeType `json:"volumeTypes,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
+}
+
+// KubernetesSettings contains constraints regarding allowed values of the 'kubernetes' block in the Shoot specification.
+type KubernetesSettings struct {
+	// Versions is the list of allowed Kubernetes versions with optional expiration dates for Shoot clusters.
+	// +patchMergeKey=version
+	// +patchStrategy=merge
+	// +optional
+	Versions []ExpirableVersion `json:"versions,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
+}
+
+// MachineImage defines the name and multiple versions of the machine image in any environment.
+type MachineImage struct {
+	// Name is the name of the image.
+	Name string `json:"name"`
+	// Versions contains versions and expiration dates of the machine image
+	// +patchMergeKey=version
```

Comment:
Same question about the patchMergeKey?

Reply:
I don't know, please contact @mvladev. I just took what he contributed for `v1alpha1`. Let's do it outside of this PR.